### PR TITLE
Fix `load_from_tensor` with newer PyTorch versions

### PR DESCRIPTION
**Pull request by araffin**
## Description

Fixes failure on SB3-Contrib master.

Note: the error happens only when a `th.Tensor` is passed instead of a `np.array`.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: stable_baselines3/common/policies.py [hunk 1]
```diff
@@ -199,7 +199,7 @@ def load_from_vector(self, vector: np.ndarray) -> None:
 
         :param vector:
         """
-        th.nn.utils.vector_to_parameters(th.FloatTensor(vector, device=self.device), self.parameters())
+        th.nn.utils.vector_to_parameters(th.as_tensor(vector, dtype=th.float, device=self.device), self.parameters())
```
Comment (qgallouedec) [inline stable_baselines3/common/policies.py:202]: Actually, I'm pretty sure that this method is wrongly type hinted. It should be 

```diff
- def load_from_vector(self, vector: np.ndarray) -> None:
+ def load_from_vector(self, vector: th.Tensor) -> None:
```

Consequently, I wonder if the following could work:

```python
th.nn.utils.vector_to_parameters(vector.to(self.device), self.parameters())

Comment (araffin) [reply stable_baselines3/common/policies.py:202]: > Actually, I'm pretty sure that this method is wrongly type hinted. It should be

it is meant for evolution strategies, so converting candidates (numpy arrays) to parameters of a policy.

In practice, with ARS, I might use the GPU for matrix multiplication too, so probably should be `Union[np.ndarray, th.Tensor]` (I also need to check what I'm doing with CEM).

Comment (araffin) [reply stable_baselines3/common/policies.py:202]: @qgallouedec sorry for the ping, but I would like to have this one merge soon as ARS is currently broken...

Comment (qgallouedec) [reply stable_baselines3/common/policies.py:202]: no problem, approving as a temporary fix

**Review by qgallouedec — APPROVED** [review]

**Pull request merged**